cibuildwheel/oci_container.py

```diff
@@ -226,13 +227,14 @@ def __enter__(self) -> Self:
                     *run_cmd, *platform_args, self.image, *ctr_cmd, capture_stdout=True
                 ).strip()
             except subprocess.CalledProcessError:
-                # The image might have been built with amd64 architecture
-                # Let's try that
-                platform_args = self._get_platform_args(oci_platform=OCIPlatform.AMD64)
-                container_machine = call(
-                    *run_cmd, *platform_args, self.image, *ctr_cmd, capture_stdout=True
-                ).strip()
-            simulate_32_bit = container_machine != "i686"
+                if self.oci_platform == OCIPlatform.i386:
+                    # The image might have been built with amd64 architecture
+                    # Let's try that
+                    platform_args = self._get_platform_args(oci_platform=OCIPlatform.AMD64)
+                    container_machine = call(
+                        *run_cmd, *platform_args, self.image, *ctr_cmd, capture_stdout=True
+                    ).strip()
+            simulate_32_bit = container_machine not in {"i686", "armv7l", "armv8l"}
```

Comment:
What's `armv8l` doing here? Remnant from testing or something required?

Reply:
this comes from the fact that a 32 bit kernel will report `armv7l` but a 64 bit kernel running linux32 emulation in an armv7 image will report `armv8l`.